### ISSUE TITLE
✨ Enhancement: It would be great having a Github Action that creates a PR with the changes required after a Add new tool issue is created

### ISSUE THREAD

**Issue by benjagm** [json-schema-org/website#1133]
### Is your feature request related to a problem? Please describe

Everytime that an "Add a new tool" issue is created, we need to manually add the changes while this can be easily automated.

### Describe the solution you'd like

I'd like to have an action that triggers after the creation of the issue and creates a PR with the required changes.

### Are you working on this?

No

**Comment by benjagm** [comment]
What do you think @aialok @DarhkVoyd ?

**Comment by rishabhknowss** [comment]
Hey @benjagm! I would like to create this. Could you clarify which changes need to be done manually and are repetitive?

**Comment by DarhkVoyd** [comment]
*benjagm* This would be great to have.
@rishabhknowss The tools data for the [/tools](https://json-schema.org/tools) page is available here: [here](https://github.com/json-schema-org/website/blob/main/data/tooling-data.yaml). Currently, adding a new tool to the data involves creating an [issue](https://github.com/json-schema-org/website/issues/new?assignees=&labels=Status%3A+Triage&projects=&template=adding-your-tooling.yml&title=Add+or+Modify+%5BTool+Name%5D+in+JSON+Schema+Ecosystem) manually and waiting for someone to pick it up and create a pull request (PR). However, this process can be automated by using a GitHub Action that creates a PR with the necessary changes to add the tools. Hope this helps!

**Comment by rishabhknowss** [comment]
> _benjagm_ This would be great to have. @rishabhknowss The tools data for the [/tools](https://json-schema.org/tools) page is available here: [here](https://github.com/json-schema-org/website/blob/main/data/tooling-data.yaml). Currently, adding a new tool to the data involves creating an [issue](https://github.com/json-schema-org/website/issues/new?assignees=&labels=Status%3A+Triage&projects=&template=adding-your-tooling.yml&title=Add+or+Modify+%5BTool+Name%5D+in+JSON+Schema+Ecosystem) manually and waiting for someone to pick it up and create a pull request (PR). However, this process can be automated by using a GitHub Action that creates a PR with the necessary changes to add the tools. Hope this helps!

Okay working on this issue !

**Comment by Sahil-Shadwal** [comment]
hey @rishabhknowss are you working on this cause i have almost solved this, please let me know .

**Comment by rishabhknowss** [comment]
Yup, guys, I am currently working on this. I will push the changes by evening! You can also contribute by suggesting any changes you think are necessary ( much appreciated )

**Comment by Sahil-Shadwal** [comment]
Hey  @benjagm , @DarhkVoyd

i have few doubts, i have done this to solve this issue:

- a workflow (.github/workflows/auto-add-tool.yml) that triggers on issues.opened, runs a script (scripts/parseIssue.ts), and creates a PR.
- the script parses the issue body, updates tooling-data.yaml, and downloads the logo (if provided).

is this the right approach ?

**Comment by rishabhknowss** [comment]
Ah, I got the time to work on this! While I was solving this issue, I simply created a GitHub Action that triggers when an issue is created and saves the issue's information directly to the tooling-data.json file. After that, it creates a commit.
![image](https://github.com/user-attachments/assets/8aab2772-c708-45a2-abc3-195064537dfb)
Something like this ? What do you think, guys? I guess not using another script would be a better approach here @Sahil-Shadwal 
Can anyone tell me how to test the GitHub Actions locally? Meanwhile, I am fixing the workflow code!!

**Comment by MadhavDhatrak** [comment]
hi 

i work on the this workflow create-tool-pr.yml This GitHub Actions workflow is designed to automate the process of adding a new tool to the ```data/tooling-data.yaml ``` file whenever an issue is triggered with the "Add Tool" label.

Key Features:
  - Trigger: The workflow is activated when an issue is opened or edited with the "Add Tool" label.
  - Branch Creation: Automatically creates a new branch named after the tool and the current timestamp.
  - Data Addition: Appends the extracted tool information to the data/tooling-data.yaml file.
  - Commit and Push: Commits the changes and pushes the new branch to the repository.
  - Pull Request Creation: Automatically creates a pull request for the new tool addition, summarizing the changes made.
  
  **Video Demo**

https://github.com/user-attachments/assets/9e35570d-7fd5-4389-985f-f6f930c5f87a

**Comment by MadhavDhatrak** [comment]
@DhairyaMajmudar  @benjagm  Can you assign me to this issue To make a PR?

**Comment by DhairyaMajmudar** [comment]
@MadhavDhatrak still up to take this ?

**Comment by Vishv0407** [comment]
@DhairyaMajmudar can I work on this issue?, since no one is working and I have basic understanding of the process because just opened a PR #1494

**Comment by MadhavDhatrak** [comment]
@DhairyaMajmudar  Yes,I am still interested this Please assign me and I'll make a PR soon

**Comment by vtushar06** [comment]
Hii @DhairyaMajmudar, if this issue available, love to take this up.

**Comment by vishwansh01** [comment]
Hey @DhairyaMajmudar @Utkarsh-123github 
**Approach:**
Create a .yml file.
This GitHub Actions setup helps automate the process of inserting a new tool into the data/tooling-data.yaml file whenever an issue is tagged with the "Add Tool" label.
Main Functions:
 - PR Automation: It then opens a pull request that outlines the details of the tool being added.
 - Activation: The workflow kicks in whenever an issue is created or modified and carries the "Add Tool" label.
 - Versioning: The workflow commits these updates and pushes the newly formed branch.
 - Branch Setup: It spins up a fresh branch that includes the tool's name along with a timestamp.
 - Data Update: The tool information collected from the issue is added to the data/tooling-data.yaml file.

I would also like to take up this issue, as it seems really exciting to me.
Let me know if any refinements are needed!

**Comment by manya1632** [comment]
Hi @benjagm  @DhairyaMajmudar ! This enhancement makes a lot of sense and could save quite a bit of repetitive manual work. I’d be happy to help implement a GitHub Action for this if the issue is still available — please let me know or assign it to me.
